src/data/D2ApiIndicator.ts

```diff
@@ -238,4 +85,223 @@ export class D2ApiIndicator {
+        );
+    }
+
+    private buildIndicatorsFromGroups(
+        dataElementGroup: D2DataElementGroup,
+        config: D2Config
+    ): Indicator[] {
+        return _(dataElementGroup.dataElements)
+            .compactMap((dataElement): Maybe<Indicator> => {
+                return this.buildOutputIndicator(dataElementGroup, dataElement, config);
+            })
+            .value();
+    }
+
+    private buildOutputIndicator(
+        dataElementGroup: D2DataElementGroup,
+        dataElement: D2DataElementFromGroup,
+        config: D2Config
+    ): Maybe<Indicator> {
+        const isOutPut = dataElement.dataElementGroups.some(
+            deg => deg.id === config.dataElementGroups.outputIndicator.id
+        );
+        if (!isOutPut) return undefined;
+
+        const scope = this.getScope(dataElement.dataElementGroups, config, "dataElementGroups");
+        if (!scope) return undefined;
+
+        const theme = dataElement.dataElementGroups.find(deg =>
+            deg.groupSets.find(gs => gs.id === config.dataElementGroupSets.theme.id)
+        );
+
+        const status = dataElement.dataElementGroups.find(deg =>
+            deg.groupSets.find(gs => gs.id === config.dataElementGroupSets.status.id)
+        );
+
+        const group = dataElement.attributeValues.find(
+            attribute => attribute.attribute.id === config.attributes.group.id
+        );
+
+        return Indicator.create({
+            denominator: "",
+            numerator: "",
+            coreCompetency: {
+                id: dataElementGroup.id,
+                code: dataElementGroup.code,
+                name: dataElementGroup.displayName,
+            },
+            id: dataElement.id,
+            name: dataElement.displayName,
+            code: dataElement.code,
+            theme: this.getValueOrEmpty(theme?.displayName),
+            status: this.getValueOrEmpty(status?.displayName),
+            type: "outputs",
+            scope: scope,
+            group: this.getValueOrEmpty(group?.value),
+            disaggregation: dataElement.categoryCombo
+                ? {
+                      id: dataElement.categoryCombo.id,
+                      name: dataElement.categoryCombo.displayName,
+                      categories: convertToCategories(dataElement.categoryCombo.categories),
+                  }
+                : undefined,
+            relatedDataElements: [],
+            categories: [],
+        });
+    }
 }
+
+type D2IndicatorGroup = {
+    id: Id;
+    code: string;
+    displayName: string;
+    indicators: Array<{
+        code: string;
+        attributeValues: Array<{ attribute: { id: Id }; value: string }>;
+        denominator: string;
+        displayName: string;
+        id: Id;
+        indicatorGroups: Array<{
+            displayName: string;
+            id: Id;
+            indicatorGroupSet: { id: Id; displayName: string };
+        }>;
+        numerator: string;
+    }>;
+};
+
+type D2DataElementGroup = {
```

Comment:
We have MetadataPick, but nothing wrong with being explicit (just more verbose), it's equally type-safe.